Client/Frontend/Library/HistoryPanel/HistoryPanelViewModel.swift

```diff
@@ -291,55 +278,26 @@ class HistoryPanelViewModel: Loggable, FeatureFlaggable {
-            return profile.history.getSitesByLastVisit(limit: queryFetchLimit, offset: currentFetchOffset) >>== { result in
-                // Force 100ms delay between resolution of the last batch of results
-                // and the next time `fetchData()` can be called.
-                DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(100)) {
-                    self.isFetchInProgress = false
-                    self.browserLog.debug("currentFetchOffset is: \(self.currentFetchOffset)")
-                }
-                return deferMaybe(result)
+        let deferred = profile.places.getSitesWithBound(
+            limit: queryFetchLimit,
+            offset: currentFetchOffset,
+            excludedTypes: VisitTransitionSet(0)
+        )
+
+        let ret = Deferred<Maybe<Cursor<Site>>>()
+        deferred.upon { sites in
+            ret.fill(sites)
+            guard sites.isSuccess else {
+                self.isFetchInProgress = false
+                return
             }
-        case .new:
-            let deferred = profile.places.getVisitPageWithBound(
-                limit: queryFetchLimit,
-                offset: currentFetchOffset,
-                excludedTypes: VisitTransitionSet(0)
-            )
-            let ret = Deferred<Maybe<Cursor<Site>>>()
-            deferred.upon { result in
-                guard let result = result.successValue else {
-                    // It's possible for our query to be interrupted
-                    // either by triggering the interrupt or by the user leaving putting the application
-                    // in the background
-                    // we should make sure we re-set the fetching flag
-                    self.isFetchInProgress = false
-                    ret.fill(Maybe(failure: result.failureValue ?? "Unknown Error"))
-                    return
-                }
-                DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(100)) {
-                    self.isFetchInProgress = false
-                    self.browserLog.debug("currentFetchOffset is: \(self.currentFetchOffset)")
-                }
-                let sites = result.infos.map { info -> Site in
-                    var title: String
-                    if let actualTitle = info.title, !actualTitle.isEmpty {
-                        title = actualTitle
-                    } else {
-                        // In case there is no title, we use the url
-                        // as the title
-                        title = info.url
-                    }
-                    let site = Site(url: info.url, title: title)
-                    site.latestVisit = Visit(date: UInt64(info.timestamp) * 1000)
-                    return site
-                }.uniqued()
-                ret.fill(Maybe(success: ArrayCursor(data: sites)))
+            DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(100)) {
+                self.isFetchInProgress = false
```

Comment:
Could be worth keeping this comment ? Since we still have the 100ms delay
```
                // Force 100ms delay between resolution of the last batch of results
                // and the next time `fetchData()` can be called.
```